### PR TITLE
Convert loading component to HOC

### PR DESCRIPTION
**Pull request by 9gustin**
## Summary

Convert loading component to HOC

## Screenshots

-

## Trello Card

[Patterns II - HOCs](https://trello.com/c/WohsDPH5/52-patterns-ii-hocs)

### CODE REVIEW

File: react/w-books/src/app/components/Loading/index.tsx [hunk 1]
```diff
@@ -5,19 +5,39 @@ import styles from './styles.module.scss';
...
+      <svg width="64px" height="64px" viewBox="0 0 128 128" className={clsx(
+        styles.loading, {
+        [styles.isGreen]: isGreen
+      }
+      )}>
+        <g>
+          <path className={styles.iconPath} d="M75.4 126.63a11.43 11.43 0 0 1-2.1-22.65 40.9 40.9 0 0 0 30.5-30.6 11.4 11.4 0 1 1 22.27 4.87h.02a63.77 63.77 0 0 1-47.8 48.05v-.02a11.38 11.38 0 0 1-2.93.37z" fill="#000000" fill-opacity="1" />
+          <animateTransform attributeName="transform" type="rotate" from="0 64 64" to="360 64 64" dur="1000ms" repeatCount="indefinite"></animateTransform>
+        </g>
+      </svg>
```
Comment (LucasZibell) [inline react/w-books/src/app/components/Loading/index.tsx:23]: This should be a svg file

File: react/w-books/src/app/screens/Book/index.tsx [hunk 2]
```diff
@@ -39,8 +39,13 @@ export default function Book({ match: urlParams }: BookProps) {
...
         {
-          loading ? (<div className={styles.container}> <Loading isGreen /> </div>)
-            : book && <Card className={styles.bookCard} {...bookToCard(book)} translateProps/>
+          withLoading(Card)({
+            loading,
+            loadingClassName: styles.container,
+            className: styles.bookCard,
+            translateProps: true,
+            ...bookToCard(book)
+          })
```
Comment (LucasZibell) [inline react/w-books/src/app/screens/Book/index.tsx:48]: Use it as a hook, not inside the component

File: react/w-books/src/app/screens/Home/index.tsx [hunk 3]
```diff
@@ -32,11 +32,14 @@ export default function Home() {
       {error && (<ErrorMessage className={styles.container}>{t(`Home:${error.problem}`)}</ErrorMessage>)}
 
       {
-        loading ? (<div className={styles.container}> <Loading isGreen /> </div>)
-          : <CardList items={response?.page ? response.page.map(bookToCard) : []} />
+        withLoading(CardList)({
+          loading,
+          loadingClassName: styles.container,
+          items: response?.page ? response.page.map(bookToCard) : []
+        })
```
Comment (LucasZibell) [inline react/w-books/src/app/screens/Home/index.tsx:39]: same

File: react/w-books/src/app/screens/Login/components/LoginForm/index.tsx [hunk 4]
```diff
@@ -77,11 +77,15 @@ export default function LoginForm({ testId, handleLogin }: LoginFormProps) {
...
         {
-          loading ? (<div className={styles.loading}> <Loading isGreen /> </div>)
-            : <Button isFilled isSubmit>{t('Common:LoginButton')}</Button>
+          withLoading(Button)({
+            loading,
+            loadingClassName: styles.loading,
+            isFilled:true, 
+            isSubmit:true,
+            children: t('Common:LoginButton'),
+          })
```
Comment (LucasZibell) [inline react/w-books/src/app/screens/Login/components/LoginForm/index.tsx:86]: same

File: react/w-books/src/app/screens/SignUp/components/SignUpForm/index.tsx [hunk 5]
```diff
@@ -116,8 +116,13 @@ export default function SignUpForm({testId, handleSignUp}:SignUpFormProps) {
...
         {
-          loading ? (<div className={styles.loading}> <Loading isGreen /> </div>)
-            : <Button isFilled isSubmit>{t('Common:SignUpButton')}</Button>
+          withLoading(Button)({
+            loading,
+            loadingClassName: styles.loading,
+            isFilled:true, 
+            isSubmit:true,
+            children: t('Common:SignUpButton'),
+          })
```
Comment (LucasZibell) [inline react/w-books/src/app/screens/SignUp/components/SignUpForm/index.tsx:125]: same

File: react/w-books/src/utils/bookToCard.ts [hunk 6]
```diff
@@ -24,5 +24,5 @@ export function bookToCard(book: Book): iCard {
         value: `${book.year}`,
       }
     ]
-  };
+  }:{};
```
Comment (LucasZibell) [inline react/w-books/src/utils/bookToCard.ts:27]: check eslint

**Review by LucasZibell — CHANGES_REQUESTED** [review]

**Comment by 9gustin** [comment]
changes are ready for review

**Review by DavidRojas1612 — APPROVED** [review]

**Pull request merged**